### ISSUE TITLE
WINE_MONO_VERBOSE=2 causes "* Assertion: should not be reached at ... wine-mono-4.7.3/mono/mono/mini/helpers.c:297"

### ISSUE THREAD

**Issue by HinTak** [madewokherd/wine-mono#41]
I tried setting WINE_MONO_VERBOSE=1 with a bug I am looking at ( https://github.com/madewokherd/wine-mono/issues/29 ) . It seems that wine-mono is a little bit better than mono 5.12's mixed mode support, but not as good as 5.14+ as far as the app I am interested in goes.

WINE_MONO_VERBOSE=1 stops at the same place as without it; but setting to WINE_MONO_VERBOSE=2 causes wine-mono to stop much earlier, with:

```
* Assertion: should not be reached at /builddir/build/BUILD/wine-mono -4.7.3/mono/mono/mini/helpers.c:297

```

By much earlier, I mean mono 5.12 with `--mixed-mode` and `--verbose` gives a log of 345k ; wine-mono with WINE_MONO_VERBOSE=1 gives a log with 375k . while mono 5.14 and mono git current master HEAD gives a log of 470k .

while WINE_MONO_VERBOSE=2 gives a log of 1.4k ... so obviously it stops a lot earlier.

also, some debug aid doc would be useful...

**Comment by HinTak** [comment]
Wine-mono 6.0.0 no longer asserts, but it seems that it is writing so much stuff out it crashes wine's debugger. I'll keep this open, but start a new issue with the log.

**Comment by madewokherd** [comment]
Should be fixed in Wine in version 6.17 by 20dc5f378ca70f15cfe28a87e8ebda6a10eeed6f.
